### PR TITLE
Fix LRScheduler import for PyTorch 2.0

### LINKED ISSUE #15912: MisconfigurationException "MultiStepLR" with torch 2.0
### Bug description

Got following error after upgrading to torch 2.0:

lightning_lite.utilities.exceptions.MisconfigurationException: The provided lr scheduler `MultiStepLR` doesn't follow PyTorch's LRScheduler API. You should override the `LightningModule.lr_scheduler_step` hook with your own logic if you are using a custom LR scheduler.

### How to reproduce the bug

```python
https://colab.research.google.com/drive/1Nks7DPZlrxUDW0UGa_2-17uOMbH0aWle?usp=sharing
```

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 1.10):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

### More info

The error came from 

https://github.com/Lightning-AI/lightning/blob/6cc493360d9dfdd132665343d6611e66e9760885/src/pytorch_lightning/core/optimizer.py#L351

`LRSchedulerTypeTuple` is using `torch.optim.lr_scheduler._LRScheduler` to identify torch schedulers

https://github.com/Lightning-AI/lightning/blob/6cc493360d9dfdd132665343d6611e66e9760885/src/pytorch_lightning/utilities/types.py#L114

However, pytorch has changed `_LRScheduler` to be subclass of `LRScheduler` ([link](https://github.com/pytorch/pytorch/blob/7571134f69475f45e4b820cb328fcb23cd5ee299/torch/optim/lr_scheduler.py#L168))\
And `MultiStepLR` is subclass of `LRScheduler` as well ([link](https://github.com/pytorch/pytorch/blob/7571134f69475f45e4b820cb328fcb23cd5ee299/torch/optim/lr_scheduler.py#L401))

A fix could be change to use `torch.optim.lr_scheduler.LRScheduler` in `LRSchedulerTypeTuple`

### PR DESCRIPTION
**Pull request by lantiga**
## What does this PR do?

(future) PyTorch 2.0 changed the class hierarchy for `LRScheduler` so our internal checks on types fail.
The problem is well summarized in @teamclouday's issue here https://github.com/Lightning-AI/lightning/issues/15912.

This PR fixes the issue so that both PyTorch 1.x and 2.0 are covered.

Fixes #15912

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: src/pytorch_lightning/utilities/types.py [hunk 1]
```diff
@@ -27,6 +27,11 @@
 from torchmetrics import Metric
 from typing_extensions import Protocol, runtime_checkable
 
+try:
+    from torch.optim.lr_scheduler import LRScheduler as TorchLRScheduler
+except ImportError:
+    from torch.optim.lr_scheduler import _LRScheduler as TorchLRScheduler
```
Comment (awaelchli) [inline src/pytorch_lightning/utilities/types.py:33]: Let's maybe add a comment here with version information? Eventually, one day Lightning will have a minimum torch version requirement such that that the old import always fails.

Something along the lines of
```
# For torch<x.y.z
or
# TODO: Remove once minimum torch version is x.y.z
```

**Review by awaelchli — APPROVED** [review]

File: src/pytorch_lightning/utilities/types.py (hunk 1 above)
Comment (lantiga) [reply src/pytorch_lightning/utilities/types.py:33]: done
